rewrite-maven/src/test/java/org/openrewrite/maven/MavenSettingsTest.java

```diff
@@ -334,13 +340,41 @@ void serverCredentials() {
           .matches(repo -> repo.getPassword().equals("my_password"));
     }
 
+    @Test
+    void serverTimeouts() {
+        var settings = MavenSettings.parse(new Parser.Input(Paths.get("settings.xml"), () -> new ByteArrayInputStream(
+          //language=xml
+          """
+                <settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
+                    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                    xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 http://maven.apache.org/xsd/settings-1.0.0.xsd">
+                      <servers>
+                        <server>
+                          <id>server001</id>
+                          <configuration>
+                            <connectTimeout>40000</connectTimeout>
+                            <readTimeout>50000</readTimeout>
+                          </configuration>
```

Comment:
I'm having another look at https://maven.apache.org/guides/mini/guide-http-settings.html#connection-timeouts
Over there it seems there's two options
## A. an older `<timeout>` directly in `<configuration>`
```xml
<settings>
  <servers>
    <server>
      <id>my-server</id>
      <configuration>
        <timeout>6000</timeout> <!-- milliseconds -->
      </configuration>
    </server>
  </servers>
</settings>
```
## B. a newer `<connectionTimeout>` and `<readTimeout>` nested inside `<httpConfiguration><put>`
```xml
<settings>
  <servers>
    <server>
      <id>my-server</id>
      <configuration>
        <httpConfiguration>
          <put>
            <connectionTimeout>10000</connectionTimeout> <!-- milliseconds -->
          </put>
        </httpConfiguration>
      </configuration>
    </server>
  </servers>
</settings>
```

## In this PR
We seem to have added `connectTimeout` (as opposed to `connectionTimeout`, as well as `readTimeout`, but neither of those nested in the `<httpConfiguration>` + `<put>` elements I'd expect from the Maven docs.

In practice I think these fields won't be set when reading a Maven settings file from disk in style A or B above. Should we make any adjustments here @ammachado ?

Reply:
Picked up in 
- https://github.com/openrewrite/rewrite/pull/4302